Tools/serial/generate_config.py

```diff
@@ -0,0 +1,274 @@
+# Omitted default values will be set to 0.
+serial_ports = {
+    "GPS1": {
+        "label": "GPS 1",
+        "defaults": {
+            "CONFIG": 'Main GPS',
+            "BAUD": 0,
+            },
+        },
+    "GPS2": {
+        "label": "GPS 2",
+        "defaults": {
+            "CONFIG": 'Disabled',
+            "BAUD": 0,
+            },
+        },
+    "TEL1": {
+        "label": "TELEM 1",
+        "defaults": {
+            "CONFIG": 'MAVLink',
+            "BAUD": 57600,
+            "MAV_MDE": 'Normal',
+            "MAV_R": 1200,
+            "MAV_FWD": 1,
+            },
+        },
+    "TEL2": { # companion port
+        "label": "TELEM 2",
+        "defaults": {
+            "CONFIG": 'Disabled', # disabled by default to reduce RAM usage
+            "BAUD": 57600,
+            "MAV_MDE": 'OSD',
+            },
+        },
+    "TEL3": {
+        "label": "TELEM 3",
+        "defaults": {
+            "CONFIG": 'Disabled',
+            "BAUD": 57600,
+            "MAV_MDE": 'Normal',
+            },
+        },
+    "TEL4": {
+        "label": "TELEM 4",
+        "defaults": {
+            "CONFIG": 'MAVLink',
+            "BAUD": 57600,
+            "MAV_MDE": 'Normal',
+            "MAV_R": 1200,
+            },
+        },
+    "URT3": { # for Omnibus
+        "label": "UART 3",
+        "defaults": {
+            "CONFIG": 'MAVLink',
+            "BAUD": 57600,
+            "MAV_MDE": 'Normal',
+            "MAV_R": 1200,
+            "MAV_FWD": 1,
+            },
+        },
+    "URT6": { # for Omnibus
+        "label": "UART 6",
+        "defaults": {
+            "CONFIG": 'Main GPS',
+            "BAUD": 0,
+            },
+        },
+    }
+
+# Serial commands: each dict entry is a list with 2 items:
+# - user-facing label (must be short, will be shown in a drop-down list)
+# - ROMFS command (can be more than one or any scripting logic).
+#   These variables can be used:
+#       ${SER_DEV}             Serial device (e.g. /dev/ttyS1)
+#       ${SER_TAG}             Serial tag (e.g. GPS1)
+#       SER_${SER_TAG}_CONFIG  Param name for the current configuration
+#       SER_${SER_TAG}_BAUD    Param name for the current baudrate
+#       SER_${SER_TAG}_MAV_MDE Param name for the current mavlink mode
+#       SER_${SER_TAG}_MAV_R   Param name for the current mavlink rate
+#       SER_${SER_TAG}_MAV_FWD Param name for current mavlink forwarding
+#       ${DUAL_GPS_ARGS}       Arguments passed to the gps to start the secondary GPS
+#       ${MAV_ARGS}            Mavlink arguments (baudrate, mode and rate)
+#
+# Note: do NOT re-use or change indexes. When adding a command, always use an
+# index that has never been used before. This is important for compatibility
+# with QGC
+serial_commands = {
+    0: ["Disabled", ""],
+
+    # MAVLink & RTPS
+    1: ["MAVLink", "mavlink start -d ${SER_DEV} ${MAV_ARGS} -x"],
+
+    2: ["MAVLink over Syslink", '''
+syslink start
+mavlink start -d /dev/bridge0 ${MAV_ARGS}
```

Comment:
Does this one need to be configurable or can it be done automatically on a crazyflie?